addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/handler/BoxHandler.java

```diff
@@ -75,14 +77,19 @@
      * Schedule for polling
      */
     private ScheduledFuture<?> pollingJob;
+    /**
+     * shared instance of HTTP client for asynchronous calls
+     */
+    private HttpClient httpClient;
 
     /**
      * Constructor
      *
      * @param bridge Bridge object representing a FRITZ!Box
      */
-    public BoxHandler(@NonNull Bridge bridge) {
+    public BoxHandler(@NonNull Bridge bridge, @NonNull HttpClient httpClient) {
```

Comment:
Only `@NonNullByDefault` and `@Nullable` should be used when using null annotations (see [code style **12.**](https://www.eclipse.org/smarthome/documentation/development/guidelines.html#a-code-style)).

Reply:
@cweitkamp You have removed `NonNull`, but that means that no annotation is there now - did you try adding `@NonNullByDefault` on class level?

Reply:
No, I didn't try that. I think it doesn't make sense to do it in this PR. I refactored the handler structure in PR #3022 which indeed has to be updated because of some merge conflicts.

Reply:
But then you at least shouldn't touch any of the existing annotations either - with your last commit, you removed the annotation for the bridge.